### DISCUSSION TITLE
Cannot access cookie in middleware after redirecting

### DISCUSSION THREAD

**Discussion by arienzIT** [nuxt/nuxt#21245] [Questions] [answered]
Hi everyone, I'm working on an email verification process and I'm attempting to access a cookie set in an `api` route before doing a `sendRedirect` from a route middleware but it's always `undefined`. 
I can instead access it easily from the client after the navigation completes.

Here's my setup:

api
```ts
// /server/api/auth/verify.get.ts
export default defineEventHandler(async (event) => {
   // ... email verification logic
  setCookie(event, 'accessToken', accessToken, {
    secure: true,
    sameSite: 'strict',
  })
  return sendRedirect(event, '/admin')
```

page
```vue
// /pages/admin/index.vue
<script setup lang="ts">
definePageMeta({
  middleware: ['auth']
})
</script>

<template>
  <h1>admin</h1>
</template>
```

middleware
```ts
// /middleware/auth.ts
export default defineNuxtRouteMiddleware((to, from) => {
    const token = useCookie('accessToken')

    if (!token.value) {
        return navigateTo('/auth/signin')
    }
})
```

Is there anything I'm missing? If then in the `Signin` component `onMounted` I try to log the same cookie I can access its value, but from the middleware I cannot.

**Comment by v3nt** [comment]
@arienzIT did you make any progress with this?

**Reply by arienzIT** [reply]
I could get around that issue by accessing the cookie from a composable and use it instead of the `useCookie` utils.
That was not the only change I did to the overall logic though, so I’m not really confident if that was the fix

**Comment by adenvt** [comment, marked as answer]
I was struggling with this issue long time ago. That’s actually because the sameSite set to ‘strict’.

Browser won’t sent the cookie after redirection.

Check here for solution: https://stackoverflow.com/questions/42216700/how-can-i-redirect-after-oauth2-with-samesite-strict-and-still-get-my-cookies

**Reply by arienzIT** [reply]
Lovely, thanks a lot for your answer!
